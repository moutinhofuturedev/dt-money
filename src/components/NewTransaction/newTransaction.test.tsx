import { render } from '@testing-library/react'

import { NewTransactionModal } from './index'

const mockDialogContext = {
  // Inclua quaisquer valores que o componente precise do contexto
  openDialog: jest.fn(),
  closeDialog: jest.fn(),
}

// Substitua o useContext do React
jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useContext: jest.fn(() => mockDialogContext),
}))

describe('NewTransactionModal', () => {
  it('should submit the form with invalid input and display error messages', async () => {
    const { container } = render(<NewTransactionModal />)

    expect(container).toMatchSnapshot()
  })
})
