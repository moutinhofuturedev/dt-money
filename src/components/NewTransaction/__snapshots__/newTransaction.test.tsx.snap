// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewTransactionModal should submit the form with invalid input and display error messages 1`] = `<div />`;
